web: add RAC Provider to the list of providers understood by the wizard

kensternberg-authentik opened the pull request:
## Details

This commit adds the RAC Provider to the list of providers understood by the Application Wizard.

It also creates a new, simple alert that knows how to look up the enterprise requirements and chooses to fill itself in with a notice saying "A license is required for this provider," or nothing.  That harmonizes the display across both wizards, and reduces the demands on the wizards themselves to "know" about enterprise features.



## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [X] The code has been formatted (`make web`)
-   [X] The translation files have been updated (`make i18n-extract`)
    -   [X] No new text was added

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

kensternberg-authentik committed:
web: add RAC Provider to the list of providers understood by the wizard

This commit also creates a new, simple alert that knows how to look up the enterprise requirements
and chooses to fill itself in with a notice saying "A license is required for this provider," or
nothing.  That harmonizes the display across both wizards, and reduces the demands on the wizards
themselves to "know" about enterprise features.

kensternberg-authentik committed:
web: remove console.log() from ak-license-notice

kensternberg-authentik committed:
web: fix inconsistencies in identity passing.

kensternberg-authentik committed:
web: fix inconsistencies in identity passing.

kensternberg-authentik committed:
web: move the license summary information into a top-level context.

Rather than repeatedly fetching the license summary, this commit
fetches it once at the top-level and keeps it until an EVENT_REFRESH
reaches the top level.  This prevents the FOUC (Flash Of Unavailable
Content) while loading and awaiting the end of the load.

kensternberg-authentik committed:
Remove some debugging info, fix a misspelling.

kensternberg-authentik committed:
Remove some debugging info, fix a misspelling.

kensternberg-authentik committed:
web: provide a context for enterprise license status

There are a few places (currently 5) in our code where we have checks for the current enterprise
licensing status of our product. While not particularly heavy or onerous, there's no reason to
repeat those same lines, and since our UI is always running in the context of authentik, may as well
make that status a client-side context in its own right. The status will update with an
EVENT_REFRESH request.

A context-aware custom alert has also been provided; it draws itself (or `nothing`) depending on the
state of the license, and the default message, "This feature requires an enterprise license," can be
overriden with the `notice` property.

These two changes reduce the amount of code needed to manage our license alerting from 67 to 38
lines code, and while removing 29 lines from a product with 54,145 lines of code (a savings of
0.05%, oh boy!) isn't a miracle, it does mean there's a single source of truth for "Is this instance
enterprise-licensed?" that's easy to access and use.

kensternberg-authentik committed:
web: [x] The translation files have been updated

kensternberg-authentik committed:
web: add RAC Provider to the list of providers understood by the wizard

This commit also creates a new, simple alert that knows how to look up the enterprise requirements
and chooses to fill itself in with a notice saying "A license is required for this provider," or
nothing.  That harmonizes the display across both wizards, and reduces the demands on the wizards
themselves to "know" about enterprise features.

kensternberg-authentik committed:
web: move the license summary information into a top-level context.

Rather than repeatedly fetching the license summary, this commit
fetches it once at the top-level and keeps it until an EVENT_REFRESH
reaches the top level.  This prevents the FOUC (Flash Of Unavailable
Content) while loading and awaiting the end of the load.

kensternberg-authentik committed:
Merge remote-tracking branch 'refs/remotes/origin/web/add-rac-to-wizard' into web/add-rac-to-wizard

* refs/remotes/origin/web/add-rac-to-wizard:
  Remove some debugging info, fix a misspelling.
  web: move the license summary information into a top-level context.
  web: fix inconsistencies in identity passing.
  web: remove console.log() from ak-license-notice
  web: add RAC Provider to the list of providers understood by the wizard

BeryJu committed:
remmove endpoint fetch from both rac provider forms since its not used

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

BeryJu committed:
i18n

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

kensternberg-authentik committed:
Merge branch 'main' into web/add-rac-to-wizard

* main: (119 commits)
  web: provide a context for checking the status of the enterprise license (#8153)
  core: compile backend translations (#8311)
  translate: Updates for file web/xliff/en.xlf in zh-Hans (#8304)
  translate: Updates for file web/xliff/en.xlf in zh_CN (#8305)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh_CN (#8300)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in zh-Hans (#8301)
  events: fix missing labels on prometheus metrics (#8309)
  core: bump goauthentik.io/api/v3 from 3.2023106.4 to 3.2023106.5 (#8302)
  web: bump the wdio group in /tests/wdio with 4 updates (#8303)
  web: restore test anchor tag (#8298)
  translate: Updates for file web/xliff/en.xlf in fr (#8296)
  translate: Updates for file locale/en/LC_MESSAGES/django.po in fr (#8295)
  website: update wording (#8290)
  enterrpise: exclude inactive users from license (#8294)
  web: bump API Client version (#8292)
  core: compile backend translations (#8291)
  events: migrate SystemTasks to DB (#8159)
  web/admin: fix footer links not being parsed on settings page (#8289)
  root: fix system check warnings (#8277)
  web: bump API Client version (#8288)
  ...

kensternberg-authentik committed:
Merge branch 'main' into web/add-rac-to-wizard

* main:
  web: clear out selecteds list after an API event to ensure a fresh copy of the policies-to-delete list (#8125)
  web: provide dual-list multiselect with pagination (#8004)

BeryJu commented on web/src/admin/applications/wizard/methods/rac/ak-application-wizard-authentication-for-rac.ts:
```diff
@@ -0,0 +1,119 @@
+                                ${msg("Hold control/command to select multiple items.")}
+                            </p>
+                        </ak-form-element-horizontal>
+                        <ak-form-element-horizontal label=${msg("Settings")} name="settings">
```
We could probably not show this here? since ux wise it's much easier to set these settings via a mapping

BeryJu replied:
Only here is fine

kensternberg-authentik committed:
web: RAC updates

- special case: disable RAC provider in the wizard if enterprise is not enabled
- remove `settings` YAML editor from the RAC provider in the wizard
